### ISSUE TITLE
Rework YAML configuration file format

### ISSUE THREAD

**Issue by jsolbrig** [NRLMMD-GEOIPS/geoips#54]
# Requested Update

### Description
Define a new YAML configuration file format that is self-describing, consistent across interfaces, and interfaces well with the updated interfaces being developed under #37.

### Background and Motivation
The current YAML configuration file format is not well defined and is not self-describing. There are no consistent fields used across different interfaces and some fields should actually be values. This issue proposes a new YAML configuration file format that will better integrate with the plugin infrastructure that we are developing under #37 which includes a specification for required fields for plugins.

### Proposed Format
As a first crack at this, I propose that we impose the following format where everything contained under the `arguments` field is interface-specific.
```yaml
plugin:
  name:  # The name of the plugin (cannot conflict with other plugins for the same interface)
  interface:  # The interface that the plugin belongs to
  family:  # The family that the plugin belongs to
  short_description:  # A short description of the plugin for use at the command line (likely length limited)
  description:  # A long description of the plugin which will be used as the docstring. This should likely be multi-line by using the pipe "|" operator.
  arguments:
    <arguments to control the plugin would go here>
```

Any thoughts on this would be appreciated!

**Comment by jsolbrig** [comment]
### Current plans
We will be running a sprint over the next week to address this issue and, probably others. 

We have put together a draft specification for the YAML configs. We will need to agree on a final format before we can proceed with the rest of the sprint. Below is an example of a `boundaries` plugin:
```yaml
interface: boundaries
metadata:
  name: tc-pmw
  family: ...
  short-description: ...
  long-description: ...
spec:
  dict-type: 'standard'
  coastlines:
    enabled: true
    linewidth: 2
    color: '#00B4B4B4'
  countries:
    enabled: true
    linewidth: ...
    color: ...
  states:
    same as above
  rivers:
    same as above
```

### Issues for the Sprint
- [ ] Agree upon and finalize base schema
- [ ] Develop, agree upon, and finalize schema for each interface
  - [ ] `plotting_params`
  - [ ] `product_inputs`
  - [ ] `product_params`
  - [ ] `sectors_dynamic`
  - [ ] `sectors_static`
- [ ] Convert existing YAML files for each interface
- [ ] Move `interface_modules` to `plugins`
- [ ] Move `yaml_configs` to `plugins`
- [ ] Create BaseYamlInterface class
- [ ] Create interface-specific Interface classes
- [ ] Address yaml file discovery for GeoIPS and plugin repositories
- [ ] Stretch goals
  - [ ] Develop a `test` interface and convert test_data_* repositories to be test plugins
  - [ ] Develop documentation for each of the interfaces
  - [ ] Schema for data_fusion repo's interfaces
